### PR TITLE
[Misc] Improve logs and metrics across several components.

### PR DESCRIPTION
**Pull request by JoshLind**
Note: this PR is built on https://github.com/aptos-labs/aptos-core/pull/11178

### Description
This PR improves the logs, configs and metrics across several components. Each change is offered in its own commit:
1. [Network] Add a metric to the network stack to count expired but received RPC responses (i.e., RPC responses that arrive too late). Also, reduce the `info!` log to `debug!`.
2. [Storage Service] Clean up and add several metrics to the storage service to: (i) better track storage request latencies; and (ii) track latencies across successes and failures.
3. [Network] Reduce the `warn!` log for `Application layer unexpectedly dropped response channel` to `info!`. This is very noisy.
4. Increase the number of outbound network connections for fullnodes from `4` to `6`.

### Test Plan
Existing test infrastructure.

### CODE REVIEW

**Review by brianolson — APPROVED** [review]
the network/framework/ stuff already has an equivalent implemented in the network2 branch, so I'll just drop this side in any merge conflict.
storage-service stuff looks fine

File: network/framework/src/peer/mod.rs [hunk 1]
```diff
@@ -270,12 +270,29 @@ where
...
-                    if let Err(err) = self.inbound_rpcs.send_outbound_response(&mut write_reqs_tx, maybe_response).await {
-                        warn!(
-                            NetworkSchema::new(&self.network_context).connection_metadata(&self.connection_metadata),
-                            error = %err,
-                            "{} Error in handling inbound rpc request, error: {}", self.network_context, err,
-                        );
+                    if let Err(error) = self.inbound_rpcs.send_outbound_response(&mut write_reqs_tx, maybe_response).await {
+                        // It's quite common for applications to drop an RPC request.
+                        // If this happens, we want to avoid logging a warning/error
+                        // (as it makes the logs noisy). Otherwise, we log normally.
```
Comment (bchocho) [inline network/framework/src/peer/mod.rs:276]: Awesome. Noticed this as well digging into humio recently

File: state-sync/storage-service/server/src/utils.rs [hunk 2]
```diff
@@ -175,3 +177,48 @@ pub fn notify_peer_of_new_data<T: StorageReaderInterface>(
 
     Ok(transformed_data_response)
 }
+
+/// An utility that calls and times the given function. The metric histogram
+/// is updated with the given labels (e.g., peer, request result and duration).
+/// If no start time is specified, the timer begins before calling the function.
+pub fn execute_and_time_duration<T, F>(
```
Comment (bchocho) [inline state-sync/storage-service/server/src/utils.rs:184]: Seems similar in spirit (but not exact functionality) to `monitor!` in consensus. Maybe there's some unifying to do down the road?

**Review by bchocho — APPROVED** [review]
LGTM

File: state-sync/storage-service/server/src/utils.rs (hunk 2 above)
Comment (JoshLind) [reply state-sync/storage-service/server/src/utils.rs:184]: Yeah, definitely. I think storage also does its own thing here, too 😄

File: network/framework/src/peer/mod.rs [hunk 3]
```diff
@@ -270,12 +270,29 @@ where
...
+                        let network_schema = NetworkSchema::new(&self.network_context)
+                            .connection_metadata(&self.connection_metadata);
+                        let error_string = format!("{} Error in handling inbound rpc request, error: {}", self.network_context, error);
+                        match error {
+                            RpcError::UnexpectedResponseChannelCancel => {
```
Comment (grao1991) [inline network/framework/src/peer/mod.rs:281]: The problem is not just log level, but more about not providing enough useful information.. I'd like to at least know which rpc method is this error about, or I think even info! is too high..

File: state-sync/storage-service/server/src/metrics.rs [hunk 4]
```diff
@@ -127,12 +130,32 @@ pub static STORAGE_RESPONSES_SENT: Lazy<IntCounterVec> = Lazy::new(|| {
...
-/// Time it takes to fetch a storage service response
+/// Time it takes to read data from the storage service DB
+pub static STORAGE_DB_READ_LATENCY: Lazy<HistogramVec> = Lazy::new(|| {
+    register_histogram_vec!(
+        "aptos_storage_service_server_db_read_latency",
+        "Time it takes to read data from the storage service DB",
+        &["request_type", "result"],
+    )
+    .unwrap()
+});
```
Comment (grao1991) [inline state-sync/storage-service/server/src/metrics.rs:141]: bucket?

File: state-sync/storage-service/server/src/metrics.rs [hunk 5]
```diff
@@ -127,12 +130,32 @@ pub static STORAGE_RESPONSES_SENT: Lazy<IntCounterVec> = Lazy::new(|| {
...
 pub static STORAGE_FETCH_PROCESSING_LATENCY: Lazy<HistogramVec> = Lazy::new(|| {
     register_histogram_vec!(
         "aptos_storage_service_server_fetch_processing_latency",
-        "Time it takes to fetch a storage service response",
-        &["network_id", "request_type"],
+        "Time it takes to fetch and package a storage service response",
+        &["network_id", "request_type", "result"],
+    )
+    .unwrap()
+});
```
Comment (grao1991) [inline state-sync/storage-service/server/src/metrics.rs:151]: ditto

File: state-sync/storage-service/server/src/metrics.rs [hunk 6]
```diff
@@ -127,12 +130,32 @@ pub static STORAGE_RESPONSES_SENT: Lazy<IntCounterVec> = Lazy::new(|| {
...
+
+/// Time it takes to create a storage service response
+pub static STORAGE_RESPONSE_CREATION_LATENCY: Lazy<HistogramVec> = Lazy::new(|| {
+    register_histogram_vec!(
+        "aptos_storage_service_server_response_creation_latency",
+        "Time it takes to create a storage service response",
+        &["network_id", "request_type", "result"],
```
Comment (grao1991) [inline state-sync/storage-service/server/src/metrics.rs:158]: ditto

File: state-sync/storage-service/server/src/metrics.rs (hunk 4 above)
Comment (JoshLind) [reply state-sync/storage-service/server/src/metrics.rs:141]: Do you mean add custom buckets to these latencies? The defaults should already be good enough, no? i.e., none of these should go beyond 10 seconds, and I don't need that fine a granularity 😄 Or did you mean something else?

**Review by grao1991 — APPROVED** [review]

File: network/framework/src/peer/mod.rs (hunk 3 above)
Comment (JoshLind) [reply network/framework/src/peer/mod.rs:288]: FYI: I've made this `debug!` now. There's not a lot of useful information we can display without a much bigger refactor, so it makes sense to ignore this log in most cases 😄

**Pull request merged**